allennlp/tools/drop_eval.py

```diff
@@ -145,14 +152,17 @@ def answer_json_to_strings(answer: Dict[str, Any]) -> Tuple[Tuple[str, ...], str
     Takes an answer JSON blob from the DROP data release and converts it into strings used for
     evaluation.
     """
-    if answer["number"]:
+    if "number" in answer and answer["number"]:
         return tuple([str(answer["number"])]), "number"
-    elif answer["spans"]:
+    elif "spans" in answer and answer["spans"]:
         return tuple(answer["spans"]), "span" if len(answer["spans"]) == 1 else "spans"
-    else:
+    elif "date" in answer:
         return tuple(["{0} {1} {2}".format(answer["date"]["day"],
                                            answer["date"]["month"],
                                            answer["date"]["year"])]), "date"
+    else:
+        print("Answer type not found, should be one of number, spans or date at: {0}".format(json.dumps(answer)))
+        raise ValueError
```

Comment:
Minor style issue: the error message should be an argument to `ValueError()`, instead of a print statement beforehand.  Also, prefer python f-strings: `raise ValueError(f"Answer type ... date at: {json.dumps(answer)}")`.